Sources/Confidence/ConfidenceClient/RemoteConfidenceClient.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+import Common
+import os
+
+public class RemoteConfidenceClient: ConfidenceClient {
+    private var options: ConfidenceClientOptions
+    private let metadata: ConfidenceMetadata
+    private var httpClient: HttpClient
+    private var baseUrl: String
+
+    init(
+        options: ConfidenceClientOptions,
+        session: URLSession? = nil,
+        metadata: ConfidenceMetadata
+    ) {
+        self.options = options
+        switch options.region {
+        case .global:
+            self.baseUrl = "https://events.confidence.dev/v1/events"
+        case .europe:
+            self.baseUrl = "https://events.eu.confidence.dev/v1/events"
+        case .usa:
+            self.baseUrl = "https://events.us.confidence.dev/v1/events"
+        }
+        self.httpClient = NetworkClient(session: session, baseUrl: baseUrl)
+        self.metadata = metadata
+    }
+
+    func upload(batch: [ConfidenceEvent]) async throws -> Bool {
+        let timeString = Date.backport.nowISOString
+        let request = PublishEventRequest(
+            events: batch.map { event in
+                Event(
+                    eventDefinition: "eventDefinitions/\(event.name)",
+                    payload: event.payload,
+                    time: timeString
+                )
+            },
+            clientSecret: options.credentials.getSecret(),
+            sendTime: timeString,
+            sdk: Sdk(id: metadata.name, version: metadata.version)
+        )
+
+        do {
+            let result: HttpClientResult<PublishEventResponse> =
+            try await self.httpClient.post(path: ":publish", data: request)
+            switch result {
+            case .success(let successData):
+                guard successData.response.status == .ok else {
+                    throw successData.response.mapStatusToError(error: successData.decodedError)
```

Comment:
if the server returns `401..499` except it was `429` we should consider it a success (as not trying again and removing the events)